orm/encoding/ormfield/timestamp.go

```diff
@@ -1,16 +1,180 @@
 package ormfield
 
 import (
+	"fmt"
 	"io"
 
 	"google.golang.org/protobuf/reflect/protoreflect"
 )
 
-// TimestampCodec DurationCodec encodes a google.protobuf.Timestamp value as 12 bytes using
-// Int64Codec for seconds followed by Int32Codec for nanos. This allows for
-// sorted iteration.
+// TimestampCodec encodes google.protobuf.Timestamp values with the following
+// encoding:
+// - nil is encoded as []byte{0xFF}
+// - seconds (which can range from 0001-01-01T00:00:00Z to 9999-12-31T23:59:59Z) is encoded as 5 fixed bytes
+// - nanos (which can range from 0 to 999,999,999) is encoded as:
+//   - []byte{0x0} for zero nanos
+//   - 4 fixed bytes with the bit mask 0xC0 applied to the first byte
+//
+// When iterating over timestamp indexes, nil values will always be ordered last.
+//
+// Values for seconds and nanos outside the ranges specified by google.protobuf.Timestamp will be rejected.
 type TimestampCodec struct{}
 
+const (
+	timestampNilValue       = 0xFF
+	timestampZeroNanosValue = 0x0
+	timestampSecondsMin     = -62135579038
+	timestampSecondsMax     = 253402318799
+	timestampNanosMax       = 999999999
+)
+
+var (
+	timestampNilBz       = []byte{timestampNilValue}
+	timestampZeroNanosBz = []byte{timestampZeroNanosValue}
+)
+
+func (t TimestampCodec) Encode(value protoreflect.Value, w io.Writer) error {
+	// nil case
+	if !value.IsValid() {
+		_, err := w.Write(timestampNilBz)
+		return err
+	}
```

Comment:
Why nil bz timestamp if value is invalid rather than error?

Reply:
Invlalid in this case means that the timestamp is actually nil. It just means that the protoreflect.Value is not populated with anything. But that only happens when the field was unset.